### PR TITLE
[Mono.Cairo] Add OpenGL texture Surface support 

### PR DESCRIPTION
**Pull request by jpbruyere**
Add Device class.
Add GLSurface class, created for texture.
Add Surface types

### CODE REVIEW

**Comment by directhex** [comment]
build

**Comment by lewurm** [comment]
LGTM.  Can someone else review that please?

File: mcs/class/Mono.Cairo/Mono.Cairo/SurfaceType.cs [hunk 1]
```diff
@@ -44,5 +45,18 @@ public enum SurfaceType
...
+		Script,
+		Qt,
+		Recording,
+		VG,
+		GL,
+		Drm,
+		Tee,
+		Xml,
+		Skia,
+		SubSurface
```
Comment (alexischr) [inline mcs/class/Mono.Cairo/Mono.Cairo/SurfaceType.cs]: Looks like you're mirroring an enum which I don't see in my copy of cairo. Could you point it to us as a comment?

Comment (jpbruyere) [inline mcs/class/Mono.Cairo/Mono.Cairo/SurfaceType.cs]: That's `cairo_surface_type` enum from `src/cairo.h` in the cairo source tree, I'm on the master branch of Cairo. Don't merge if no one else show interest in having GL surface bindings. It's not actively developped.

Comment (wolfgangmauer) [inline mcs/class/Mono.Cairo/Mono.Cairo/SurfaceType.cs]: Hi,
i'm interesting on GLES2! So this seems to be the first step.

**Comment by jpbruyere** [comment]
This is my work-in-progress, on kms stack.

https://github.com/jpbruyere/Crow/blob/TestKMS/testDrm/src/DrmDevice.cs

**Comment by luhenry** [comment]
@monojenkins commit apidiff and csproj

**Pull request merged**